core/src/main/java/com/linecorp/armeria/server/file/StreamingHttpFile.java

```diff
@@ -223,7 +224,7 @@ private void doRead(HttpResponseWriter res, T in, long offset, long end,
 
                     final HttpFileBuilder builder =
                             HttpFile.builder(array != null ? HttpData.wrap(array)
-                                                           : new ByteBufHttpData(buf, true),
+                                                           : PooledHttpData.wrap(buf).withEndOfStream(),
```

Comment:
Question: Should we add `withEndOfStream(boolean)` which returns `this` if `endOfStream` did not change?

Reply:
Yeah seems good